tinymce-angular-component/src/main/ts/editor/editor.component.ts

```diff
@@ -64,15 +65,27 @@ export class EditorComponent extends Events implements AfterViewInit, ControlVal
   @Input() public modelEvents = 'change input undo redo';
   @Input() public allowedEvents?: string | string[];
   @Input() public ignoreEvents?: string | string[];
+  @Input()
+  public set readonly(val) {
+    this._readonly = val;
+    if (this._editor && this._editor.initialized) {
+      setMode(this._editor, val ? 'readonly' : 'design');
+    }
+  }
+
+  public get readonly() {
+    return this._readonly;
+  }
+
   @Input()
   public set disabled(val) {
     this._disabled = val;
     if (this._editor && this._editor.initialized) {
-      if (typeof this._editor.mode?.set === 'function') {
-        this._editor.mode.set(val ? 'readonly' : 'design');
-      } else if ('setMode' in this._editor && typeof this._editor.setMode === 'function') {
-        this._editor.setMode(val ? 'readonly' : 'design');
+      if (DisabledUtils.isDisabledOptionSupported()) {
+        this._editor.options.set('disabled', val ?? false);
+        return;
       }
+      setMode(this._editor, val ? 'readonly' : 'design');
```

Comment:
I think this line should be removed, setting the mode now should be only done when setting the `readonly` prop

Reply:
@hamza0867 are you suggesting that it should use `editor.options.set('readonly', <false|true>)` api instead?